### ISSUE TITLE
Enhancement: update `ast-spec` snapshots for `@babel/parser`

### ISSUE THREAD

**Issue by sosukesuzuki** [typescript-eslint/typescript-eslint#6265]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

https://github.com/babel/babel/pull/15209 has been merged so we should update accessor properties tests in `ast-spec`.

### Fail

```typescript
// N/A
```

### Pass

```typescript
// N/A
```

### Additional Info

- https://github.com/typescript-eslint/typescript-eslint/pull/5926#issuecomment-1319478746
